src/PrestaShopBundle/Install/Install.php

```diff
@@ -892,208 +893,92 @@ public function configureShop(array $data = [])
         $modules = [];
-
-        foreach ($addons_modules as $addons_module) {
-            if (file_put_contents(_PS_MODULE_DIR_ . $addons_module['name'] . '.zip', Tools::addonsRequest('module', ['id_module' => $addons_module['id_module']]))) {
-                if (Tools::ZipExtract(_PS_MODULE_DIR_ . $addons_module['name'] . '.zip', _PS_MODULE_DIR_)) {
-                    $modules[] = (string) $addons_module['name']; //if the module has been unziped we add the name in the modules list to install
-                    unlink(_PS_MODULE_DIR_ . $addons_module['name'] . '.zip');
-                }
+        foreach (scandir(_PS_MODULE_DIR_, SCANDIR_SORT_NONE) as $module) {
+            if ($module[0] != '.' && is_dir(_PS_MODULE_DIR_ . $module) && file_exists(_PS_MODULE_DIR_ . $module . '/' . $module . '.php')) {
+                $modules[] = $module;
             }
         }
 
-        return count($modules) ? $this->installModules($modules) : true;
+        return $modules;
     }
 
     /**
      * PROCESS : installModules
      * Download module from addons and Install all modules in ~/modules/ directory.
      *
-     * @param array|string|null $module Module to install. If not provided, installs all modules.
-     *
      * @return bool
      */
-    public function installModules($module = null)
+    public function installModules(): bool
     {
-        if ($module && !is_array($module)) {
-            $module = [$module];
+        $modules = $this->getModulesList();
+
+        Module::updateTranslationsAfterInstall(false);
+
+        $result = $this->executeEvent($modules, 'install', 'Cannot install module "%module%"');
+        if ($result === false) {
+            return false;
         }
 
-        $modules = $module ? $module : $this->getModulesList();
+        Module::updateTranslationsAfterInstall(true);
+        EntityLanguage::updateModulesTranslations($modules);
 
-        Module::updateTranslationsAfterInstall(false);
+        return true;
+    }
 
+    public function postInstall(): bool
+    {
+        return $this->executeEvent(
+            $this->getModulesList(),
+            'postInstall',
+            'Cannot execute post install on module "%module%"'
+        );
+    }
+
+    protected function executeEvent(array $modules, string $event, string $errorMessage): bool
+    {
         $moduleManagerBuilder = ModuleManagerBuilder::getInstance();
         $moduleManager = $moduleManagerBuilder->build();
 
         $errors = [];
         foreach ($modules as $module_name) {
-            if (!file_exists(_PS_MODULE_DIR_ . $module_name . '/' . $module_name . '.php')) {
-                continue;
-            }
-
             $moduleException = null;
 
             try {
-                $moduleInstalled = $moduleManager->install($module_name);
-            } catch (\PrestaShopException $e) {
-                $moduleInstalled = false;
+                $moduleEventIsExecuted = $moduleManager->{$event}($module_name);
```

Comment:
May be check if method 'pre'.$event ou 'post'.$event exist and execute it

Reply:
No hook for event

Reply:
What do you mean by "no hook for event"? Please use a proper sentence to be understandable :smile:

Reply:
We don't need to check if the method exists as we control the call, and this is not a public method.